NEXT-25671 - Fixed setting of custom field number type

LiaraAlis opened the pull request:
Fixed [NEXT-25671](https://issues.shopware.com/issues/NEXT-25671) by switching order of parameters when merging custom field config in `applyTypeConfiguration()`. This prevents overriding values set by user with default values.

Before building js and completing the tasks below, I want to discuss if this change can cause some unwanted side effects. I've tested multiple types and was not able to find any problems. With my change it was possible to create a custom field of type number with integer type. If that is okay, I can commit changes with the built js and created changelog file.

What do you think?

- [x] I have rebased my changes to remove merge conflicts
- [ ] I have written tests and verified that they fail without my change
- [x] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-implement-new-changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.


[NEXT-25671]: https://shopware.atlassian.net/browse/NEXT-25671?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

LiaraAlis committed:
Fixed NEXT-25671

Fixed NEXT-25671 by switching order of parameters when merging custom field config in `applyTypeConfiguration()`. This prevents overriding values set by user with default values.

seggewiss commented:
I think you are right by changing the orders of the attributes. Good find 👍 
Feel free to complete this MR 😊

LiaraAlis committed:
Added changelog

LiaraAlis commented:
> I think you are right by changing the orders of the attributes. Good find 👍 Feel free to complete this MR 😊

Thanks for your feedback! I added changelog and changed title to met the requirements.

LiaraAlis commented:
@seggewiss Is something mandatory missing that I need to do to proceed with this pr?

LiaraAlis commented:
Is there anything else I need to do to get this PR merged?

seggewiss commented:
@LiaraAlis I'm sincerely sorry that I overlooked this PR for so long. 😞 I'm importing it right away 👍

ghost commented:
Hello,

thank you for creating this pull request.
I have opened an issue on our Issue Tracker for you. See the issue link: https://issues.shopware.com/issues/NEXT-33091

Please use this issue to track the state of your pull request.

seggewiss commented:
@LiaraAlis Are you able to rebase this PR? Our import tool generates 6000+ changes because the branch is so old.

LiaraAlis commented:
@seggewiss Rebase done.

ghost commented:
Hello,

thank you for creating this pull request.
I have opened an issue on our Issue Tracker for you. See the issue link: https://issues.shopware.com/issues/NEXT-33091

Please use this issue to track the state of your pull request.

seggewiss commented:
Thank you for your contribution 💙
